packages/aws-cdk/lib/api/util/sdk.ts

```diff
@@ -105,6 +124,25 @@ export class SDK implements ISDK {
       });
     }
 
+    if (options.endpoints) {
+      try {
+        this.endpoints = JSON.parse(options.endpoints);
+      } catch (_e) {}
+
+      try {
+          this.endpoints = this.endpoints ||
+          options.endpoints.split(/,/g).reduce<ServiceEndpoints>((acc, entry) => {
+              const [service, url] = entry.split('=');
+              if (service && url) {
+                  acc[service] = url;
+              }
+              return acc;
+          }, {});
+      } catch (_e) {}
```

Comment:
align with style of exceptions in reset of this file.

Why swallow up the exception? Should we be logging something here?